Library Catalog (Encore).js

```diff
@@ -36,204 +35,183 @@
-				tagValue = marc.subfieldDelimiter+"a"+tagValue;
-			}
-			
-			// add previous tag
-			record.addField(tag, ind, tagValue);
-		}
-		
-		record.translate(newItem);
-		//the library catalogue name isn't perfect, but should be unambiguous. 
-		var domain = newDoc.location.href.match(/https?:\/\/([^/]+)/);
-		newItem.repository = domain[1].replace(/encore\./, "")+" Library Catalog";
-		// there is too much stuff in the note field - or file this as an abstract?
-		newItem.notes = [];		
-		newItem.complete();
-	}
-}
+				if (tagValue) {
+					tagValue = tagValue.replace(/\|(.)/g, marc.subfieldDelimiter + "$1");
+					if (tagValue[0] != marc.subfieldDelimiter) {
+						tagValue = marc.subfieldDelimiter + "a" + tagValue;
+					}
+
+					// add previous tag
+					record.addField(tag, ind, tagValue);
+				}
+
+				record.translate(newItem);
 
-function pageByPage(marc, urls) {
-	Zotero.Utilities.processDocuments(urls, function(newDoc) {
-		scrape(marc, newDoc);
-	}, function() { Zotero.done() });
+				newItem.repository = domain[1].replace(/encore\./, "");
+				// there is too much stuff in the note field - or file this as an abstract?
+				newItem.notes = [];
+				newItem.complete();
+			});
+		});
+	}
 }
 
 /** BEGIN TEST CASES **/
-var testCases = [
-	{
+var testCases = [{
 		"type": "web",
-		"url": "http://encore.colorado.edu/iii/encore/search?formids=target&lang=eng&suite=def&reservedids=lang%2Csuite&submitmode=&submitname=&target=thelen&Search.x=0&Search.y=0",
+		"url": "http://sallypro.sandiego.edu/iii/encore/search/C__Rb3558162__Stesting__Orightresult__U__X6?lang=eng&suite=cobalt#resultRecord-b3558162",
 		"items": "multiple"
 	},
 	{
 		"type": "web",
-		"url": "http://encore.coalliance.org/iii/encore/search/C|Sthelen|Orightresult|U1?lang=eng",
-		"items": "multiple"
-	},
-	{
-		"type": "web",
-		"url": "http://encore.colorado.edu/iii/encore/record/C__Rb5910060__Sthelen__P0%2C3__Orightresult__X4?lang=eng&suite=cobalt",
-		"items": [
-			{
-				"itemType": "book",
-				"creators": [
-					{
-						"firstName": "Marcel",
-						"lastName": "Thelen",
-						"creatorType": "editor"
-					},
-					{
-						"firstName": "F.",
-						"lastName": "Steurs",
-						"creatorType": "editor"
-					}
-				],
-				"notes": [],
-				"tags": [
-					"Language and languages",
-					"Terms and phrases"
-				],
-				"seeAlso": [],
-				"attachments": [],
-				"ISBN": "9789027223371",
-				"title": "Terminology in everyday life",
-				"place": "Amsterdam, The Netherlands : Philadelphia, Pa",
-				"publisher": "John Benjamins Pub. Co",
-				"date": "2010",
-				"numPages": "271",
-				"series": "Terminology and lexicography research and practice",
-				"seriesNumber": "v. 13",
-				"callNumber": "P305 .T4437 2010",
-				"libraryCatalog": "colorado.edu Library Catalog"
-			}
-		]
+		"url": "http://sallypro.sandiego.edu/iii/encore/record/C__Rb1516899__Stesting__P0%2C2__Orightresult__U__X6?lang=eng&suite=cobalt",
+		"items": [{
```

Comment:
of course tests are from scaffold, yes. I think this was jsbeautify which I ran on the whole file.

Reply:
Ah, okay. For test it makes IMO sense to leave the formatting as Scaffold outputs it. Otherwise this will show differences the next time someone is updating and testing the same translator.